### PR TITLE
#7355 Fix bug with copying mod with optional integrations

### LINKED ISSUE #7355: Error copying mod with optional integrations
## Describe the bug

When copying a mod in the Page Editor, an error is thrown but the mod is still copied. Need to confirm if the copy is accurate.

## To Reproduce

1. Create a mod that uses the Google Drive Integration
2. In the Workshop, edit the integration definition (services key) to make the integration optional 
3. In the Page Editor, make a copy of the mod

## Actual behavior

An error alert appears: `Error Creating Mod: Integration google/oauth2-pkce is not configured`. The mod is still copied.

## Expected behavior

No error is shown. The mod is copied correctly.

## Additional context

- https://pixiebrix.slack.com/archives/C0436P48QHY/p1705435019540419

### PR DESCRIPTION
**Pull request by BLoe**
## What does this PR do?

- Fixes #7355 

## Discussion

See here for extensive discussion of these changes: https://www.notion.so/pixiebrix/inferConfiguredModIntegrations-function-should-not-throw-errors-at-all-ca10961ead98475d8b0ff572e1e0b9a2

## Checklist

- [x] Add tests
- [ ] New files added to `src/tsconfig.strictNullChecks.json` (if possible)
- [x] Designate a primary reviewer - @grahamlangford

### CODE REVIEW

File: src/store/extensionsUtils.ts [hunk 1]
```diff
@@ -35,66 +35,25 @@ export function inferRecipeOptions(
...
+ * Gather configured integration dependencies from existing mod-component-like
  * instances for reinstalling a mod. Filters out any optional integrations that
  * don't have a config set.
  * @param modComponents mod components from which to extract integration dependencies
- * @param optional don't check integration dependencies for valid configs
  * @returns IntegrationDependency[] the configured integration dependencies for the mod components
  * @see installMod
  */
-export function inferConfiguredModIntegrations(
+export function gatherConfiguredIntegrationDependencies(
```
Comment (twschiller) [inline src/store/extensionsUtils.ts:45]: NIT: have been using `collect*` prefix v.s gather: https://github.com/pixiebrix/pixiebrix-extension/blob/c2a742057f210b0ee777e160b7b8a56d074a78d5/src/analysis/analysisVisitors/blockIdVisitor.ts#L48-L48

File: src/integrations/util/gatherExistingConfiguredDependenciesForMod.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import { type IntegrationDependency } from "@/integrations/integrationTypes";
+import { collectConfiguredIntegrationDependencies } from "@/store/extensionsUtils";
+import getUnconfiguredComponentIntegrations from "@/integrations/util/getUnconfiguredComponentIntegrations";
+
+/**
+ * Gather the existing configured dependencies for a given mod definition
+ * @param modDefinition the mod definition for which to gather configured dependencies
+ * @param modComponents the mod components from which to gather the configured dependencies
+ */
+function gatherExistingConfiguredDependenciesForMod(
```
Comment (grahamlangford) [inline src/integrations/util/gatherExistingConfiguredDependenciesForMod.ts:29]: NIT: `gather` -> `collect`

File: src/background/modUpdater.ts [hunk 3]
```diff
@@ -203,7 +201,7 @@ export function deactivateMod(
   reduxState: ActivatedModState,
 ): {
   reduxState: ActivatedModState;
-  deactivatedModComponents: UnresolvedModComponent[];
+  deactivatedModComponents: ActivatedModComponent[];
```
Comment (grahamlangford) [inline src/background/modUpdater.ts:204]: How are `deactivatedModComponents` of type `ActivatedModComponent[]`?

Comment (BLoe) [reply src/background/modUpdater.ts:204]: There's a lot of complexity and weirdness right now around our mod component types, and I think fixing it right now would expand the scope of this PR too much. Do you mind if I address this in a follow up DX refactoring PR instead?

Comment (BLoe) [reply src/background/modUpdater.ts:204]: We don't have a `DeactivatedModComponent` type, and these deactivated components are only used to collect the integration dependencies and the mod options, the `active` flag is not checked, and the badly-named components aren't stored anywhere or sent anywhere down stream from this mod updater functionality.

Comment (grahamlangford) [reply src/background/modUpdater.ts:204]: As discussed in slack, a follow up DX ticket would be good. For now we should just add a clarifying comment and move on.

**Review by grahamlangford — APPROVED** [review]

**Pull request merged**